src/backends/tensorrt/plan_backend.cc

```diff
@@ -980,12 +980,13 @@ PlanBackend::Context::Run(
     LOG_VERBOSE(1) << "Context with profile " << citr->second.profile_name_
                    << " [" << std::to_string(citr->first)
                    << "] is being executed for " << name_;
-    if (is_dynamic_) {
-      if (!citr->second.context_->allInputDimensionsSpecified()) {
-        return Status(
-            RequestStatusCode::INTERNAL,
-            "failed to specify the dimensions of all input bindings");
-      }
+    if (is_dynamic_ &&
+        (!citr->second.context_->allInputDimensionsSpecified())) {
+      return Status(
+          RequestStatusCode::INTERNAL,
+          "failed to specify the dimensions of all input bindings");
+    }
+    if (!engine_->hasImplicitBatchDimension()) {
```

Comment:
Is `hasImplicitBatchDimension()` going to have the same meaning as `is_dynamic_`?

Reply:
No. They are two different things but slightly related. is_dynamic_ captures whether any engine dimension contained a wildcard(dynamic shapes) whereas hasImplicitBatchDimension() captures whether engine contains an implicit batch specification. For having a dynamic shape, the network can not have implicit batch specification. However, it is possible to have an explicit batch dimension with fixed shape networks.

Have a look at the following issue:
https://github.com/NVIDIA/tensorrt-inference-server/issues/859